### PR TITLE
Codecov integration and badge

### LINKED ISSUE #133: Add code coverage badge
This project has 100% code coverage. That's probably a thing worth highlighting in the readme badges — gives potential users some confidence.

### PR DESCRIPTION
**Pull request by davidtheclark**
Closes #133.

I have never hooked up codecov before. I think this should do the trick.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
-  - "8"
-  - "10"
 git:
   depth: 5
 cache: yarn
 branches:
   only:
     - master
-    - v5
+matrix:
```
Comment (sudo-suhas) [inline .travis.yml]: Could you please tell me the significance of using matrix?

File: package.json [hunk 2]
```diff
@@ -17,6 +17,7 @@
     "test": "jest --coverage",
     "test:watch": "jest --watch",
     "coverage": "jest --coverage --coverageReporters=html --coverageReporters=text",
+    "report-coverage": "jest --coverage --coverageReporters=lcov && codecov -t d6ade9c1-c2b4-4ab1-82f4-b2feed929ca4",
```
Comment (sudo-suhas) [inline package.json]: Is that a token for Codecov?

File: .travis.yml (hunk 1 above)
Comment (davidtheclark) [reply .travis.yml]: Matrix allows us to do different things in different Travis jobs. My understanding is that defining multiple versions of Node is effectively shorthand for defining a Matrix. But in this PR I expanded that into the longhand so I could add another job that performs the same test but sends code coverage results.

File: package.json (hunk 2 above)
Comment (davidtheclark) [reply package.json]: Yes, that is the project-specific token.

**Comment by davidtheclark** [comment]
> Is there any reason this won't work for us?

That doesn't appear to be the method that is documented and suggested: https://github.com/codecov/example-node

**Comment by davidtheclark** [comment]
I was able to remove the extra job from Travis and use one of the existing jobs to report coverage, because I realized that the existing jobs were already each generating coverage data.

I think the reason to still keep the Matrix, instead of a single `after_success` hook, is because we only need one of the jobs to report coverage, not all of them.

**Comment by sudo-suhas** [comment]
@davidtheclark See https://docs.codecov.io/docs/about-the-codecov-bash-uploader.  Since codecov is a CI specific dependency, I'd prefer just using the bash script and not add a dev-dependency.

_If I am being too finicky about this, please go ahead and merge it_.

File: .travis.yml [hunk 3]
```diff
@@ -1,13 +1,14 @@
...
+  include:
+    - node_js: "4"
+    - node_js: "6"
+    - node_js: "8"
+      after_success: "npm run report-coverage"
```
Comment (sudo-suhas) [inline .travis.yml]: Since we use yarn on CI, change to `yarn report-coverage`?

**Comment by davidtheclark** [comment]
> See https://docs.codecov.io/docs/about-the-codecov-bash-uploader.

Ah, thanks! Works like a charm.

**Comment by sudo-suhas** [comment]
My pleasure @davidtheclark!

**Pull request merged**